app/models/user.rb

```diff
@@ -3,6 +3,12 @@ class User < ApplicationRecord
   has_many :comments
   has_many :likes
 
+  # Validation for name
+  validates :name, presence: true
+
+  # Validation for post_counter
+  validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
```

Comment:
- [ ] Kindly add a test for this, `recent_posts` custom method in your `User ` model to ensure that it works as expected.